src/ComposerFileStorage.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Tuf\ComposerIntegration;
+
+use Composer\Config;
+use Composer\Util\Filesystem;
+use Tuf\Client\DurableStorage\FileStorage;
+
+/**
+ * Defines storage for the PHP-TUF metadata of a repository in a Composer project.
+ */
+class ComposerFileStorage extends FileStorage
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function __construct(string $basePath)
+    {
+        (new Filesystem())->ensureDirectoryExists($basePath);
+        parent::__construct($basePath);
+    }
+
+    /**
+     * Returns the base path where data for all repositories is stored.
+     *
+     * @param Config $config
+     *   The current Composer configuration.
+     *
+     * @return string
+     *   The base path where data is stored.
+     */
+    public static function basePath(Config $config): string
+    {
+        return implode(DIRECTORY_SEPARATOR, [
+            rtrim($config->get('vendor-dir'), DIRECTORY_SEPARATOR),
+            'composer',
+            'tuf',
+        ]);
+    }
+
+    /**
+     * Creates a storage object for a specific repository.
+     *
+     * @param string $url
+     *   The repository URL.
+     * @param Config $config
+     *   The current Composer configuration.
+     *
+     * @return static
+     *   The storage object.
+     */
+    public static function create(string $url, Config $config): self
+    {
+        $basePath = implode(DIRECTORY_SEPARATOR, [
+            static::basePath($config),
+            preg_replace('/[^[:alnum:]\.]/', '-', $url),
```

Comment:
This will still have same problem

> I don't think we can just replace everything with .
> 
> site.coop.info/packages and site.coop/info/packages would both be converted to site.coop.info.packages
> so probably need to either have folder everything before first / or replace / with something else. maybe -

now it would just both be `site-coop-info-packages`. The point I was trying to make was avoid this collisions you need to replace `.` and `/` with different characters. Not the same character.

Also because the example domains that are conflicting are `coop.info` and `site.coop` this a collision that could between organization not something 1 organization could solve by using a different subdomain or path.  

I person could have repo from `site.coop.info/packages` and then later the `site.coop/info/packages` and they could pick up the wrong root.json unless I missing something

Reply:
Good point -- but this is actually what we had before this PR too. See https://github.com/php-tuf/composer-integration/blob/main/src/TufValidatedComposerRepository.php#L54. (In fact, this dates all the way back to before I even started working on this plugin.) So IMHO we should open a follow-up issue to work on that and refine it, but not block on this. What do you think?

Reply:
Or, since we're talking about a persistent storage directory, which is used under the hood, we could absolutely just switch to using a SHA-256 hash, which would get around this.

Reply:
Another point to consider: this regex (or something very similar) is, I believe, used by Composer when building cache directories for package metadata: https://github.com/composer/composer/blob/master/src/Composer/Repository/ComposerRepository.php#L134

So if they're not worried about collisions, I'm not either. That said, I'd rather just switch to using a hash of the URL as the directory name, so we can put this collision issue to bed forever (or at least until SHA-256 is broken :))

Reply:
to me using the hash would obscure it more than necessary. say if a new root.json was add to an existing if the directory was something meaningful then it would easier to understand look at changes in git or just looking at the directory structure. With the hash it would not be clear which repository the root.json.

Yeah lets make a follow up to solve this. It should a 0.1.0 or at least 1.0.0 blocker as this not something want to change after people are using this.

Reply:
Opened #24 to discuss/solve this.